osf/models/node.py

```diff
@@ -2976,6 +2968,19 @@ def save(self, *args, **kwargs):
         return super(Collection, self).save(*args, **kwargs)
 
 
+def remove_addons(auth, resource_object_list):
+    for config in AbstractNode.ADDONS_AVAILABLE:
+        try:
+            settings_model = config.node_settings
+        except LookupError:
+            settings_model = None
+
+        if settings_model:
+            addon_list = settings_model.objects.filter(owner__in=resource_object_list)
```

Comment:
[`get_addon`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/mixins.py#L232) checks `settings_model.deleted`, should we be filtering by `deleted=False` here?

Reply:
If the settings_model is deleted, it will be None. There is no is deleted field for node_settings.

Reply:
`NodeSettings` does have a `deleted` field -- it comes from [`BaseAddonSettings`](https://github.com/CenterForOpenScience/osf.io/blob/15c35127b52db0414c67451dcc6fc8b8555216cf/addons/base/models.py#L41)

Reply:
fixed in 35c19c5